example/RuntimeComponent/CMakeLists.txt

```diff
@@ -28,3 +28,16 @@ target_link_libraries(RuntimeComponent
 )
 
 enable_midlrt(RuntimeComponent)
```

Comment:
Thanks for your attention to this change! I should have clarified my scenario better up front. While I used the in-project `RuntimeComponent` target to demonstrate the feature, my real use case is generating projections for various NuGet packages which get installed into my project. In that case, I need to avoid duplicating the SDK projection for each package, and have them naturally link together with proper include paths, etc. I tried to make the dependency management as generic as possible to avoid precluding deeper hierarchies of components, but in practice my current usage is currently flat; i.e. 1-N packages which each only depend on the SDK itself.

I did notice some of the related requirements in `enable_midlrt` as well as the potential investment/improvement for robust hierarchies of components, etc. I specifically didn't try to address any of those in my change since I'm not actually using `enable_midlrt` at all in my current project. The improvements you suggest all sound good though; I'm just not the use-case to drive those requirements. :)

One thing that I did consider though, as you suggested above, was propagating `CPPWINRT_VERSION` through a target property, since all dependencies will need to use the same version to be compatible. That makes it simpler to configure as well; you just set it locally on the target that generates your SDK projection and everything else refers to that in the normal CMake way.

Reply:
Oh. Gotcha.